### ISSUE TITLE
Fedora 27 build

### ISSUE THREAD

**Issue by robnagler** [radiasoft/zgoubi#28]
We had a conversation about portability and the outcome was that we would not build a portable install system. Our initial focus is RadiaSoft's deployment environment:  Fedora 27, gfortran, openmpi, etc. 

You can get this environment by pulling our docker image radiasoft/beamsim or you can use vagrant on a Mac:

```sh
curl radia.run | bash -s vagrant-sirepo-dev
```

Simplicity is in the eye of the beholder so I understand your desire to use your build environment. Unfortunately, it's very complicated for me to read. I think the build script could be something like:

```sh
#!/bin/bash
rm -rf build
mkdir build
cd build
cmake  --prefix="$(pyenv prefix)" ..
```

I tested this, and it seems to work fine (still with the same errors).

**Comment by robnagler** [comment]
I forgot to mention that our build environment assumes pyenv. This is a minor detail, but it is important to us to put all binaries in the current pyenv install prefix.

**Comment by zbeekman** [comment]
Do we want to use this docker image for CI testing? 

FWIW, right now things look pretty portable, at least macOS and linux.

**Comment by robnagler** [comment]
Compiles and tests pass on F27. 

Yes, we need to use Docker on Travis. You can use [radiasoft/beamsim-part1:latest](https://hub.docker.com/r/radiasoft/beamsim-part1/) which has the environment that zgoubi needs.

Portability may seem free, but it's not. For example, we want to be integrated with pyenv. The build script is currently written in Bash 3.x which was released in 2006. If you write modern Bash, you really want to be using Bash 4.

**Comment by robnagler** [comment]
I didn't give enough context. The reason the scripts are written in Bash 3 is that [Apple cannot release a newer version due to licensing issues](https://apple.stackexchange.com/a/208405). Just like old FORTRAN is not good FORTRAN, we don't want to be held back by Apple's licensing woes.

As to portability, we can write portable FORTRAN, and we should. We do not have to write portable installers or build systems. There are too many of them. RadiaSoft's devops philosophy is clear: we deploy Fedora images on whatever operating system you like. Here are some of the reasons:

* Development, in particular devops, is much easier and faster when only one operating system needs to be considered.
* Execution is more reproducible when the operating system and tools are tightly constrained.
* Simple systems are more robust, more scalable, and easier to debug

**Comment by rouson** [comment]
I just added this issue the SOW 1 milestone so as not to lose track of it.  Pull request #33 provides a proposed fix for this issue.

**Comment by zbeekman** [comment]
@robnagler FYI I took over working on this issue and the fix-issue-#28-fedora-27-build from @rouson. This is related to #10 and #27. Currently, I have implemented Travis-CI testing only using the radiasoft/beamsim-part1:latest docker image, as discussed in [this comment on #10](https://github.com/radiasoft/zgoubi/issues/10#issuecomment-407562472)
